### PR TITLE
[034][FE] Integrate to sidebar all information of user 

### PR DESCRIPTION
**Pull request by jasonchuacodes**
## Asana Link

https://app.asana.com/0/1203011106874563/1203087509751364

## Note
  - [ ] This task incorporated the changes based on client request. The university table now has a User foreignId. 
  - [ ] Please run the migration and seeders before testing the changes
  - [ ] The RouteList is updated. Outlet is used so that Sidebar only renders once to prevent multiple unnecessary request to BE.

## Pre-requisite

  - [ ] Setup postgres database
  - [ ] Add data by running the seeders

## Commands

  - [ ] in the terminal, go to the project directory and cd into api directory - `cd api`
  - [ ] run `php artisan migrate:fresh --seed`
  - [ ] in the terminal, go to the project directory and cd into api directory - `cd client`
  - [ ] run `npm i`
  - [ ] run `npm start`

## Testing
* As admin
  - [ ] the admin has access to ALL the university colleges > departments > and courses
  - [ ] login as admin by clicking on the Login button in the dashboard
  - [ ] input the ff credentials email: `admin@ebulletin.com` pass: `password`
  - [ ] navigate to the home page and test by clicking on the items in the Sidebar
  #### Expected Output
  - [ ] Upon clicking the collegeCard, it should show the available departments under that college
  - [ ] Upon clicking the departmentCard, it should show the available courses/classes under that department
  - [ ] styling is also set to indicate which card is active

#### Screenshots
image

* As student
  - [ ] the student can only view the college>department he is enrolled in and all the courses/classes he took.
  - [ ] login as user by clicking on the Login button in the dashboard
  - [ ] input the ff credentials email: `user@ebulletin.com` pass: `password`
  - [ ] navigate to the home page and test by clicking on the items in the Sidebar
  ####  Expected Output
  - [ ] Upon clicking the collegeCard, it should show the department the user is enrolled in
  - [ ] Upon clicking the departmentCard, it should show the available courses/classes the user is taking
  - [ ] styling is also set to indicate which card is active

#### Screenshots
image

### CODE REVIEW

**Comment by JohnRoyCabezas** [comment]
Kindly update the sidebar icons to match the ones we have in our `Admin Settings Page`. Thanks!
image

File: api/app/Http/Controllers/UserController.php [hunk 1]
```diff
@@ -2,15 +2,33 @@
...
 use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Auth;
 
 class UserController extends Controller
 {
-    //
+    public function getAuthUser()
+    {
+        $user = Auth::user();
+        $id = Auth::id();
```
Comment (ejanton-sun) [inline api/app/Http/Controllers/UserController.php:18]: I think you can get the id from the `$user`

File: api/app/Http/Controllers/UserController.php [hunk 2]
```diff
@@ -2,15 +2,33 @@
...
+
+        if ($user->roleUser->role_id === 2) {
```
Comment (ejanton-sun) [inline api/app/Http/Controllers/UserController.php:20]: create scope for this 
Ref: https://laravel.com/docs/9.x/eloquent#local-scopes
`$user->student`

File: api/app/Http/Controllers/UserController.php [hunk 3]
```diff
@@ -2,15 +2,33 @@
...
+            $user = User::with('university.colleges.departments.courses')->find($id);
+            return response()->json($user);
```
Comment (ejanton-sun) [inline api/app/Http/Controllers/UserController.php:22]:
```suggestion
```
remove

File: api/app/Http/Controllers/UserController.php [hunk 4]
```diff
@@ -2,15 +2,33 @@
...
+        } else {
+            $user = User::with('department.college.university', 'courseUsers.course')->find($id);
+            return response()->json($user);
```
Comment (ejanton-sun) [inline api/app/Http/Controllers/UserController.php:25]: remove `return response()->json($user);`
```suggestion
```

File: api/app/Http/Controllers/UserController.php [hunk 5]
```diff
@@ -2,15 +2,33 @@
...
+        }
+    }
```
Comment (ejanton-sun) [inline api/app/Http/Controllers/UserController.php:27]: Put return statement at the end of the condition
```suggestion
        }
        return response()->json($user);
    }
```

File: client/src/components/CollegeAccordion.js [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+  data,
+  departments,
+  department,
+}) => {
+  const [isActive, setIsActive] = useState(false);
+  const user = JSON.parse(Cookies.get('user') || '{}');
+  const ROLES = {
+    STUDENT: 1,
+    ADMIN: 2,
+  };
```
Comment (ejanton-sun) [inline client/src/components/CollegeAccordion.js:17]: This can be on separete JSON file

**Review by ejanton-sun — CHANGES_REQUESTED** [review]

File: client/src/components/CollegeAccordion.js (hunk 6 above)
Comment (jasonchuacodes) [reply client/src/components/CollegeAccordion.js:17]: Comment applied.

File: api/app/Http/Controllers/UserController.php (hunk 5 above)
Comment (jasonchuacodes) [reply api/app/Http/Controllers/UserController.php:27]: Comment applied.

File: api/app/Http/Controllers/UserController.php (hunk 4 above)
Comment (jasonchuacodes) [reply api/app/Http/Controllers/UserController.php:25]: Comment applied.

File: api/app/Http/Controllers/UserController.php (hunk 3 above)
Comment (jasonchuacodes) [reply api/app/Http/Controllers/UserController.php:22]: Comment applied.

File: api/app/Http/Controllers/UserController.php (hunk 2 above)
Comment (jasonchuacodes) [reply api/app/Http/Controllers/UserController.php:20]: Comment unresolved. Unable to use scope to return truthy or falsy values.

File: api/app/Http/Controllers/UserController.php (hunk 1 above)
Comment (jasonchuacodes) [reply api/app/Http/Controllers/UserController.php:18]: Comment applied.

File: api/app/Http/Controllers/UserController.php (hunk 2 above)
Comment (jasonchuacodes) [reply api/app/Http/Controllers/UserController.php]: Comment resolved.

**Review by NiloJr-sun — APPROVED** [review]

**Pull request merged**